lib/models/application.coffee

```diff
@@ -146,6 +158,7 @@ getApplicationModel = (deps, opts) ->
 					if size(applications) > 1
 						throw new errors.ResinAmbiguousApplication(nameOrId)
 				.get(0)
+				.tap normalizeDeviceInfo
```

Comment:
I'd probably refactor the `normalizeDeviceInfo` out, so we only do it once. I think if you put a single `tap` unindented just above `asCallback`, it should handle both cases in one go.